automation/TaskManager.py

```diff
@@ -111,6 +111,23 @@ def __init__(self, manager_params, browser_params, process_watchdog=False):
 
         self._save_configuration(browser_params)
 
+        # read the last used site visit id
+        cur = self.db.cursor()
+        last_visit_id = 0
+        query_successful = False
+        while not query_successful:
+            try:
+                cur.execute("SELECT MAX(visit_id) from site_visits")
+                self.db.commit()
+                last_visit_id = cur.fetchone()[0]
+                if last_visit_id is None:
+                    last_visit_id = 0
+                query_successful = True
+            except OperationalError:
+                time.sleep(0.1)
+                pass
+        self.next_visit_id = last_visit_id + 1
+
```

Comment:
The only reason to wrap in the `while not query successful` loop is to avoid `OperationalError`s that arise from multiple writers attempting to write to the database at the same time. That error shouldn't be thrown on a read. Also, there is no need to commit after a SELECT.

Can be reduced to:
```
# read the last used site visit id
cur = self.db.cursor()
cur.execute("SELECT MAX(visit_id) from site_visits")
last_visit_id = cur.fetchone()[0]
if last_visit_id is None:
    last_visit_id = 0
self.next_visit_id = last_visit_id + 1
```